test/components/Provider.spec.js

```diff
@@ -312,5 +312,43 @@ describe('React', () => {
       ReactDOM.unmountComponentAtNode(div)
       expect(spy).toHaveBeenCalledTimes(1)
     })
+
+    it('should handle store and children change in a the same render', () => {
```

Comment:
I did verify that this did actually fail on the old code before making my change.